Details about known caveats of using multiple processes in pre-commit (#9463)
pylint-dev/pylint: doc/user_guide/usage/run.rst

be checked in parallel. Discovered problems by checkers are not displayed
immediately. They are shown just after checking a module is complete.

You can also do you own parallelization by launching pylint multiple time on subsets
of your files (like ``pre-commit`` with the default ``require_serial=false`` does).
Be aware though pylint should analyses all your code at once in order to best infer
the actual values that are passed in calls. If only some of the files are given pylint
might miss a particular value's type and produce inferior inference for the subset.
It can also be unexpectedly different when the file set changes because the new
slicing can change the inference. So, don't do this if correctness and determinism is

> important to you.

Author: I'm not sure if ``-j`` option is really better than a custom attempt at parallelization atm.

> You can also do you own parallelization by launching pylint multiple time on subsets
> of your files (like ``pre-commit`` with the default ``require_serial=false`` does).
> Be aware though pylint should analyses all your code at once in order to best infer
> the actual values that are passed in calls. If only some of the files are given pylint
> might miss a particular value's type and produce inferior inference for the subset.
> It can also be unexpectedly different when the file set changes because the new
> slicing can change the inference. So, don't do this if correctness and determinism is
> important to you.

Reviewer: ```suggestion
You can also do your own parallelization by launching pylint multiple times on subsets
of your files (like ``pre-commit`` with the default ``require_serial=false`` does).
Be aware, though: pylint should analyse all your code at once in order to best infer
the actual values that result from calls. If only some of the files are given, pylint
might miss a particular value's type and produce inferior inference for the subset.
It can also be unexpectedly different when the file set changes because the new
slicing can change the inference. So, don't do this if correctness and determinism
are important to you.
```
